lib/ansible/module_utils/network/ios/config/static_routes/static_routes.py

```diff
@@ -0,0 +1,490 @@
+                  to the desired configuration
+        """
+
+        commands = []
+
+        # Drill each iteration of want n have and then based on dest and afi tyoe comparison take config call
+        for w in want:
+            for addr_want in w.get('address_families'):
+                for route_want in addr_want.get('routes'):
+                    check = False
+                    for h in have:
+                        for addr_have in h.get('address_families'):
+                            for route_have in addr_have.get('routes'):
+                                if route_want.get('dest') == route_have.get('dest')\
+                                        and addr_want['afi'] == addr_have['afi']:
+                                    check = True
+                                    have_set = set()
+                                    new_hops = []
+                                    for each in route_want.get('next_hops'):
+                                        want_set = set()
+                                        new_dict_to_set(each, [], want_set, 0)
+                                        new_hops.append(want_set)
+                                    new_dict_to_set(addr_have, [], have_set, 0)
+                                    commands.extend(self._set_config(w, h, addr_want,
+                                                                     route_want, route_have, new_hops, have_set))
+                            if check:
+                                break
+                        if check:
+                            break
+                    if not check:
+                        # For configuring any non-existing want config
+                        new_hops = []
+                        for each in route_want.get('next_hops'):
+                            want_set = set()
+                            new_dict_to_set(each, [], want_set, 0)
+                            new_hops.append(want_set)
+                        commands.extend(self._set_config(w, {}, addr_want, route_want, {}, new_hops, set()))
+
+        return commands
+
+    def _state_overridden(self, want, have):
+        """ The command generator when state is overridden
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+
+        commands = []
+        # Creating a copy of want, so that want dict is intact even after delete operation
+        # performed during override want n have comparison
+        temp_want = copy.deepcopy(want)
+
+        # Drill each iteration of want n have and then based on dest and afi tyoe comparison take config call
+        for h in have:
+            for addr_have in h.get('address_families'):
+                for route_have in addr_have.get('routes'):
+                    check = False
+                    for w in temp_want:
+                        for addr_want in w.get('address_families'):
```

Comment:
it would add more complexity if moved in common func as internally there are few changes for each of operation state, so would prefer of keeping the compute in each operation state